plugins/inputs/xtremio/xtremio.go

```diff
@@ -0,0 +1,324 @@
+	fields["volumes_read_latency"], _ = strconv.Atoi(gjson.Get(resp, "content.rd-latency").Str)
+	fields["volumes_write_latency"], _ = strconv.Atoi(gjson.Get(resp, "content.wr-latency").Str)
+	fields["volumes_data_reduction_ratio"], _ = strconv.ParseFloat(gjson.Get(resp, "content.data-reduction-ratio").Str, 64)
+	fields["volumes_provisioned_space"], _ = strconv.Atoi(gjson.Get(resp, "content.vol-size").Str)
+	fields["volumes_used_space"], _ = strconv.Atoi(gjson.Get(resp, "content.logical-space-in-use").Str)
+
+	acc.AddFields("xio", fields, tags)
+}
+
+func (xio *XtremIO) GatherXMS(wg *sync.WaitGroup, url string, acc telegraf.Accumulator) {
+	defer wg.Done()
+	resp, _ := xio.Call(url)
+	tags := map[string]string{
+		"guid":    gjson.Get(resp, "content.guid").Str,
+		"name":    gjson.Get(resp, "content.name").Str,
+		"version": gjson.Get(resp, "content.version").Str,
+		"xms_ip":  gjson.Get(resp, "content.xms-ip").Str,
+	}
+	fields := make(map[string]interface{})
+	fields["xms_write_iops"], _ = strconv.Atoi(gjson.Get(resp, "content.wr-iops").Str)
+	fields["xms_read_iops"], _ = strconv.Atoi(gjson.Get(resp, "content.rd-iops").Str)
+	fields["xms_overall_efficiency_ratio"], _ = strconv.ParseFloat(gjson.Get(resp, "content.overall-efficiency-ratio").Str, 64)
+	fields["xms_ssd_space_in_use"], _ = strconv.Atoi(gjson.Get(resp, "content.ssd-space-in-use").Str)
+	fields["xms_ram_in_use"], _ = strconv.Atoi(gjson.Get(resp, "content.ram-usage").Str)
+	fields["xms_ram_total"], _ = strconv.Atoi(gjson.Get(resp, "content.ram-total").Str)
+	fields["xms_cpu_usage_total"], _ = strconv.ParseFloat(gjson.Get(resp, "content.cpu").Raw, 64)
+	fields["xms_write_latency"], _ = strconv.Atoi(gjson.Get(resp, "content.wr-latency").Str)
+	fields["xms_read_latency"], _ = strconv.Atoi(gjson.Get(resp, "content.rd-latency").Str)
+	fields["xms_user_accounts_count"], _ = strconv.Atoi(gjson.Get(resp, "content.num-of-user-accounts").Raw)
+
+	acc.AddFields("xio", fields, tags)
+}
+
+func (xio *XtremIO) Call(endpoint string) (string, error) {
+	client := &http.Client{}
+	req, err := http.NewRequest("GET", xio.URL+"/api/json/v3/types/"+endpoint, nil)
+	if err != nil {
+		return "", err
+	}
+	req.Header.Set("Content-Type", "application/json")
+	req.Header.Set("Accept", "application/json")
+	req.AddCookie(xio.Cookie)
+	resp, err := client.Do(req)
+	if err != nil {
+		return "", err
+	}
+	defer resp.Body.Close()
+	data, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return "", err
+	}
+	return string(data), nil
+}
+
+func (xio *XtremIO) Authenticate() error {
+	client := &http.Client{}
+	req, err := http.NewRequest("GET", xio.URL+"/api/json/v3/commands/login?password="+xio.Password+"&user="+xio.Username, nil)
+	if err != nil {
+		return err
+	}
+	resp, err := client.Do(req)
+	if err != nil {
+		return err
+	}
+	for _, cookie := range resp.Cookies() {
+		if cookie.Name == "sessid" {
+			xio.Cookie = cookie
+		}
+	}
+	return nil
+}
+
+func (xio *XtremIO) ResetCookie() {
+	xio.Cookie = nil
+}
```

Comment:
I think you can do that setting directly at the calling location instead of wrapping that one line in a function.